src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1743,6 +1746,14 @@ void RadioSpinel<InterfaceType>::HandleTransmitDone(uint32_t          aCommand,
         VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
         static_cast<Mac::TxFrame *>(mTransmitFrame)->SetKeyId(keyId);
         static_cast<Mac::TxFrame *>(mTransmitFrame)->SetFrameCounter(frameCounter);
+
+#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
+        mMacFrameCounterSet = true;
+        // If we directly save the `frameCounter`, the RCP will use the `frameCounter` to send the next frame after the
+        // RCP is restored. The peer will consider a duplicate frame received and then drop the frame. Here saves the
+        // `frameCounter + 1` to avoid this case from happening.
+        mMacFrameCounter = frameCounter + 1;
```

Comment:
Should we also track/update the counter used for enhanced ack? I think we see them in `ParseRadioFrame()` as `aFrame.mInfo.mRxInfo.mAckFrameCounter`?

Reply:
I have updated the code to track the counter for enhanced ACK.